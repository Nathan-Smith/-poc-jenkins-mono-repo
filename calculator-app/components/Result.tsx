import styles from '../styles/Home.module.scss'
import { useEffect, useRef, useState } from 'react'
import { MDCTextField } from '@material/textfield'

export function Result(): JSX.Element {
  const labelRef = useRef<HTMLLabelElement>(null)
  const [value, setValue] = useState('')

  useEffect(() => {
    if (labelRef.current) new MDCTextField(labelRef.current)
  }, [labelRef])

  // Overkill number validation (/^(-?)(\d(?:\d{0,3}(?<!,),(?:\d{1,3})?)+|\d*)(?<=,\d{3}|[^,]?)((?:\.\d*))?.*/gm

  return (
    <label
      className={`mdc-text-field mdc-text-field--filled mdc-text-field--end-aligned ${styles.result}`}
      style={{ gridArea: 'result' }}
      ref={labelRef}
    >
      <span className="mdc-text-field__ripple"></span>
      <input
        data-testid="result-input"
        className={`mdc-text-field__input ${styles.resultInput}`}
        type="text"
        value={value}
        onChange={(e) => {
          const [sign = '', whole = '', decimal = '', fractional = ''] =
            e.target.value.match(/^(-?)(\d*)(\.\d*)?.*/gm) ?? []
          setValue(`${sign}${whole}${decimal}${fractional}`)
        }}
      />
      <span className="mdc-line-ripple"></span>
    </label>
  )
}
