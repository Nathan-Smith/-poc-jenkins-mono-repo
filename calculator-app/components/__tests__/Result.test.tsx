import React from 'react'
import renderer, { act, ReactTestRenderer } from 'react-test-renderer'

import { Result } from '../Result'

describe('Result', () => {
  test('renders', () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<Result value="42" />)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })

  test('update value', () => {
    let component: ReactTestRenderer | undefined
    act(() => {
      component = renderer.create(<Result value="42" />)
    })

    act(() => {
      component?.update(<Result value="69" />)
    })

    expect(component?.toJSON()).toMatchSnapshot()
  })
})
