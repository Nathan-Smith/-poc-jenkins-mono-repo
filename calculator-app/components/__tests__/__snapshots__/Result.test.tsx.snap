// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Result renders 1`] = `
<div
  className="result result"
  data-testid="result"
>
  42
</div>
`;

exports[`Result update value 1`] = `
<div
  className="result result"
  data-testid="result"
>
  69
</div>
`;
